# Data Wrangling

In [2]:
import pandas as pd
import numpy as np
from IPython.display import Image, HTML

In [3]:
## Read csv file and insert into a Dataframe

In [4]:
df = pd.read_csv('/Users/Rezalution/Desktop/the-movies-dataset/movies_metadata.csv')

/Users/Rezalution/opt/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [5]:
df.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
adult                    45466 non-null object
belongs_to_collection    4494 non-null object
budget                   45466 non-null object
genres                   45466 non-null object
homepage                 7782 non-null object
id                       45466 non-null object
imdb_id                  45449 non-null object
original_language        45455 non-null object
original_title           45466 non-null object
overview                 44512 non-null object
popularity               45461 non-null object
poster_path              45080 non-null object
production_companies     45463 non-null object
production_countries     45463 non-null object
release_date             45379 non-null object
revenue                  45460 non-null float64
runtime                  45203 non-null float64
spoken_languages         45460 non-null object
status                   45379 non-null objec

In [7]:
df.shape

(45466, 24)

There are 24 columns with 45466 rows. There are not that many NULL values though further analysis will be done to see what the actual values since some of the non-null values could be 0.

In [8]:
df = df.drop(['imdb_id','id','homepage','original_title','adult'],axis=1)

4 columns were dropped since they do not provide any value to the analysis like the IMDB ID and hoempage website.

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 19 columns):
belongs_to_collection    4494 non-null object
budget                   45466 non-null object
genres                   45466 non-null object
original_language        45455 non-null object
overview                 44512 non-null object
popularity               45461 non-null object
poster_path              45080 non-null object
production_companies     45463 non-null object
production_countries     45463 non-null object
release_date             45379 non-null object
revenue                  45460 non-null float64
runtime                  45203 non-null float64
spoken_languages         45460 non-null object
status                   45379 non-null object
tagline                  20412 non-null object
title                    45460 non-null object
video                    45460 non-null object
vote_average             45460 non-null float64
vote_count               45460 non-null flo

In [10]:
df[df.budget == '0'].shape

(36573, 19)

In [13]:
df[df.revenue == 0].shape

(38052, 19)

Both revenue and budget columns a majority of have values that equal 0. These will need to be drop since revenue and budget are both crucial features to the analysis. They both play a factor in determining the success of a movie. Therefore these values will need to be dropped. 

In [14]:
df.budget = pd.to_numeric(df.budget, errors='coerce')

In [15]:
df.budget = df.budget.replace(0,np.nan)

In [16]:
df[df.budget.isnull()].shape

(36576, 19)

In [17]:
df.revenue = df.revenue.replace(0,np.nan)

In [20]:
df[df.revenue.isnull()].shape

(38058, 19)

Both dropping all values that equaled 0 were converted to null values. Also budget was converted to numeric values so that it can be used in calculations.

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 19 columns):
belongs_to_collection    4494 non-null object
budget                   8890 non-null float64
genres                   45466 non-null object
original_language        45455 non-null object
overview                 44512 non-null object
popularity               45461 non-null object
poster_path              45080 non-null object
production_companies     45463 non-null object
production_countries     45463 non-null object
release_date             45379 non-null object
revenue                  7408 non-null float64
runtime                  45203 non-null float64
spoken_languages         45460 non-null object
status                   45379 non-null object
tagline                  20412 non-null object
title                    45460 non-null object
video                    45460 non-null object
vote_average             45460 non-null float64
vote_count               45460 non-null floa

In [33]:
df.loc[df.revenue.isnull() | df.budget.isnull()].shape

(40085, 23)

Out of the 45,466 rows only 5,381 of them have values in both the revenue and budget columns. These rows will be used for the analysis. 

In [34]:
df['profit'] = df.revenue - df.budget

In [35]:
df['profitPercentage'] = df.profit / df.budget

In [38]:
df['revenuePercentage'] = df.revenue / df.budget

3 additionally rows were created; Profit, Profit Percentage, and Revenue Percentage. These will be used to determine how successful a movie was since just using revenue alone will only show how much money it made. This does not show if there was any profit and even if it did make a profit what the percentage of that was to its actual budget. A movie can have a revenue of $10,000 and still be success because it's budget was only $5,000. While a movie that has revenue of $300,000 is not because of its $400,000 budget.

In [37]:
df.head()

belongs_to_collection      budget  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000.0   
1                                                NaN  65000000.0   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...         NaN   
3                                                NaN  16000000.0   
4  {'id': 96871, 'name': 'Father of the Bride Col...         NaN   

                                              genres original_language  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...                en   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...                en   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...                en   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...                en   
4                     [{'id': 35, 'name': 'Comedy'}]                en   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...    21.9469   
1  When siblings Judy and Peter discover an encha...    17.0155   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...    3.85949   
4  Just when George Banks has recovered from his ...    8.38752   

                        poster_path  \
0  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg   
1  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg   
2  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg   
3  /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg   
4  /e64sOI48hQXyru7naBFyssKFxVd.jpg   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  ...  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30  ...   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15  ...   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10  ...   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count       profit  \
0                    Toy Story  False          7.7     5415.0  343554033.0   
1                      Jumanji  False          6.9     2413.0  197797249.0   
2             Grumpier Old Men  False          6.5       92.0          NaN   
3            Waiting to Exhale  False          6.1       34.0   65452156.0   
4  Father of the Bride Part II  False          5.7      173.0          NaN   

   profitPercentage  revenuePercentage  \
0         11.451801          12.451801   
1          3.043035           4.043035   
2               NaN                NaN   
3          4.090760           5.090760   
4               NaN                NaN   

                                              poster  
0  <img src='http://image.tmdb.org/t/p/w185//rhIR...  
1  <img src='http://image.tmdb.org/t/p/w185//vzmL...  
2  <img src='http://image.tmdb.org/t/p/w185//6ksm...  
3  <img src='http://image.tmdb.org/t/p/w185//16XO...  
4  <img src='http://image.tmdb.org/t/p/w185//e64s...  

[5 rows x 23 columns]

In [26]:
df.shape

(45466, 22)

In [27]:
df[df.profit.isnull()].shape

(40085, 22)

In [28]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 22 columns):
belongs_to_collection    4494 non-null object
budget                   8890 non-null float64
genres                   45466 non-null object
original_language        45455 non-null object
overview                 44512 non-null object
popularity               45461 non-null object
poster_path              45080 non-null object
production_companies     45463 non-null object
production_countries     45463 non-null object
release_date             45379 non-null object
revenue                  7408 non-null float64
runtime                  45203 non-null float64
spoken_languages         45460 non-null object
status                   45379 non-null object
tagline                  20412 non-null object
title                    45460 non-null object
video                    45460 non-null object
vote_average             45460 non-null float64
vote_count               45460 non-null floa

In [29]:
posterURL = 'http://image.tmdb.org/t/p/w185/'

In [30]:
df['poster'] = "<img src='" + posterURL + df.poster_path + "' style='height:100px;'>"

In [31]:
df.poster[0]

"<img src='http://image.tmdb.org/t/p/w185//rhIRbceoE9lR4veEXuwCC2wARtG.jpg' style='height:100px;'>"

In [32]:
posterDF = pd.DataFrame(df['poster'].head())
HTML(posterDF.to_html(escape=False))

HTML code was added to the movie poster's path so that it could be displayed.